crates/volta-core/src/tool/package/manager.rs

```diff
@@ -124,15 +124,16 @@ fn get_npm_package_name(mut source_dir: PathBuf) -> Option<String> {
 fn get_single_directory_name(parent_dir: &Path) -> Option<String> {
     let mut entries = read_dir_eager(parent_dir)
         .ok()?
-        .filter_map(
-            |(entry, metadata)| {
-                if metadata.is_dir() {
-                    Some(entry)
-                } else {
-                    None
-                }
-            },
-        );
+        .filter_map(|(entry, metadata)| {
+            // If the entry is a symlink, _both_ is_dir() _and_ is_file() will be false. We want to
+            // include symlinks as well as directories in our search, since `npm link` uses
+            // symlinks internally, so we only exclude files from this search
+            if !metadata.is_file() {
```

Comment:
```suggestion
            // If the entry is a symlink, _both_ is_dir() _and_ is_file() will be false. We want to
            // include symlinks as well as directories in our search, since `npm link` uses
            // symlinks internally, so we only exclude files from this search
            if metadata.is_dir() || metadata.file_type().is_symlink() {
```